app/src/main/java/org/microg/gms/droidguard/DroidGuardDatabase.java

```diff
@@ -19,12 +19,18 @@ public DroidGuardDatabase(Context context) {
     @Override
     public void onCreate(SQLiteDatabase db) {
         // Note: "NON NULL" is actually not a valid sqlite constraint, but this is what we see in the original database *shrug*
-        db.execSQL("CREATE TABLE main (a TEXT NOT NULL, b LONG NOT NULL, c LONG NOT NULL, d TEXT NON NULL, e TEXT NON NULL,f BLOB NOT NULL,g BLOB NOT NULL);");
+        db.execSQL("CREATE TABLE IF NOT EXISTS main (a TEXT NOT NULL, b LONG NOT NULL, c LONG NOT NULL, d TEXT NON NULL, e TEXT NON NULL,f BLOB NOT NULL,g BLOB NOT NULL);");
+        db.beginTransaction();
+        db.setTransactionSuccessful();
+        db.endTransaction();
     }
 
     @Override
     public void onUpgrade(SQLiteDatabase db, int oldVersion, int newVersion) {
         db.execSQL("DROP TABLE IF EXISTS main;");
+        db.beginTransaction();
+        db.setTransactionSuccessful();
+        db.endTransaction();
```

Comment:
This should not have any effect *at all*. You are creating a transaction, setting it as successful and then finishing it, without doing anything inside that transaction.